### PR TITLE
chore(deps): update dep @testing-library/react-native from 10.0.0 to v10.1.0

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves []

### Description

Upgrade rntl and one reference I missed due to another pr getting merged.
Fixes issue from https://github.com/callstack/react-native-testing-library/issues/999
Also fixes breaking tests in main right now

### PR Checklist

- [ ] I tested my changes on **iOS** / **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- chore(deps): update dep @testing-library/react-native from 10.0.0 to v10.1.0 - gkartalis

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Review by dimatretyak — APPROVED** [review]
Nice work 🔥

File: src/app/Scenes/Artwork/Artwork.tests.tsx [hunk 1]
```diff
@@ -565,8 +565,8 @@ describe("Artwork", () => {
         }),
       })
 
-      expect(queryByA11yLabel("Create artwork alert section")).toBeNull()
-      expect(queryByA11yLabel("Create artwork alert buttons section")).toBeNull()
+      expect(queryByLabelText("Create artwork alert section")).toBeNull()
+      expect(queryByLabelText("Create artwork alert buttons section")).toBeNull()
```
[the remark this answers is not in the document]
Comment (gkartalis) [reply src/app/Scenes/Artwork/Artwork.tests.tsx:569]: Sure, this is the full [changelog](https://github.com/callstack/react-native-testing-library/releases/tag/v10.0.0) of the 10.0.0 release. Its not that clear but:

[Here](https://callstack.github.io/react-native-testing-library/docs/api-queries/#bylabeltext) you can see that now `ByLabelText` now matches the accessibilityLabel prop.

I guess they will also write a migration guide as well at some point. 

A good reference point (and it is updated) is the [How should I query](https://callstack.github.io/react-native-testing-library/docs/how-should-i-query)? section in their docs. They list every query and when it should be used.

[the remark this answers is not in the document]
Comment (dimatretyak) [reply src/app/Scenes/Artwork/Artwork.tests.tsx:569]: > Here you can see that now ByLabelText now matches the accessibilityLabel prop.

Got it. Thanks for the explanation 😅

**Review by MounirDhahri — APPROVED** [review]
Niice

**Pull request merged**